test/router/replicaset_master_uri_ipv6.test.lua

```diff
@@ -0,0 +1,25 @@
+test_run = require('test_run').new()
```

Comment:
Plese, use luatest, diff-tests are legacy, they're not used anymore

Reply:
But I just modified tests in router dir. And all test there same. Where I can find corret example of test with luatest?

Reply:
Also, the test does nothing, but checks, that the returned URI is inside `[...]`.

Does vshard cluster fails to start, when ipv6 are used?

I see, that you're trying to make the returned value not `::1:3301` but `[::1:3301]`, when ipv6 is used.
If vshard starts vshard cluster correctly with ipv6, then it's probably cartridge, which is needed to be fixed.
If vshard doesn't, then we should test, that after this change cluster operates with ipv6.

Reply:
> But I just modified tests in router dir. And all test there same. Where I can find corret example of test with luatest?

You can use router for that, the tests are inside `test/router-luatest/*` dir. But I'd recommend writing a unit test, which checks replicaset explicitly, inside `test/replicaset-luatest/replicaset_3_test.lua`

Reply:
> Does vshard cluster fails to start, when ipv6 are used?

vshard cluster starts correctly with ipv6, but cartridge uses router:info()['<replicaset-uuid>'].master.uri in `cartridge/roles/vshard-router.lua` in `bootstrap_group` to connect to router. And ::1:3301 will be incorrect. Thats why I need this changes.

Reply:
> vshard cluster starts correctly with ipv6, but cartridge uses router:info()['<replicaset-uuid>'].master.uri in cartridge/roles/vshard-router.lua in bootstrap_group to connect to router. And ::1:3301 will be incorrect. Thats why I need this changes.

Can't cartridge take `router:info()['<replicaset-uuid>'].master.uri` and add `[..]` if needed?

Reply:
Probably can, but in order to do that we need uri parser that can parse ipv6 uris without `[..]` because default `uri.parse` can't. And router still will store replicaset instance uri that inappropriate with ipv6 uri standart.

Reply:
I have to agree with @AntiFrizz1 here. IPv6 addresses need to be wrapped into `[]` before adding `:<port>`. Otherwise the address is unparsable, because both IP parts and the port are using `:` delimiter in IPv6.

Reply:
Indeed, according to [RFC 2732](https://www.rfc-editor.org/rfc/rfc2732) ipv6 must be wrapped into brackets. Uri module works wrong wit ipv6 now, created the ticket for that: https://github.com/tarantool/tarantool/issues/9556.

As we must support old Tarantool version, let's go with the change, proposed in this PR. However, let's make test with luatest, not difftest. Here it's:

```diff
diff --git a/test/replicaset-luatest/replicaset_3_test.lua b/test/replicaset-luatest/replicaset_3_test.lua
index 0107e7b..c8f8296 100644
--- a/test/replicaset-luatest/replicaset_3_test.lua
+++ b/test/replicaset-luatest/replicaset_3_test.lua
@@ -300,3 +300,14 @@ test_group.test_named_replicaset = function(g)
     t.assert_equals(ret, nil)
     vtest.storage_start(g.replica_1_b, global_cfg)
 end
+
+test_group.test_ipv6_uri = function(g)
+    local new_cfg = table.deepcopy(global_cfg)
+    local rs_uuid = g.replica_1_a:replicaset_uuid()
+    local uuid = g.replica_1_a:instance_uuid()
+    new_cfg.sharding[rs_uuid].replicas[uuid].uri = 'storage:storage@[::1]:3301'
+
+    local _, rs = next(vreplicaset.buildall(new_cfg))
+    local replica_string = 'replica_1_a(storage@[::1]:3301)'
+    t.assert_equals(tostring(rs.master), replica_string)
+end

```

Reply:
Added test with luatest. Removed previous test implementation.